### PR TITLE
VR-4714, 4719 Support computing and applying diffs for commits

### PR DESCRIPTION
**Pull request by nhatsmrt**

### CODE REVIEW

File: client/scala/src/main/scala/ai/verta/repository/Commit.scala [hunk 1]
```diff
@@ -303,4 +303,36 @@ class Commit(
...
+      ).map(r => new Diff(r.diffs))
+    }
+  }
+
+  /** Applies a diff to this Commit.
+   *  This method creates a new commit in ModelDB, and assigns a new ID to this object.
+   */
+  def applyDiff(diff: Diff, message: String)(implicit ec: ExecutionContext) = {
+    if (!saved)
+      Failure(new IllegalStateException("Commit must be saved before a diff can be applied"))
```
Comment (conradoverta) [inline client/scala/src/main/scala/ai/verta/repository/Commit.scala]:
```suggestion
      Failure(new IllegalCommitSavedStateException("Commit must be saved before a diff can be applied"))
```

File: client/scala/src/main/scala/ai/verta/repository/Commit.scala [hunk 2]
```diff
@@ -303,4 +303,36 @@ class Commit(
...
+    else
+      createCommit(message = message, diffs = diff.blobDiffs, commitBase = id)
+    }
```
Comment (conradoverta) [inline client/scala/src/main/scala/ai/verta/repository/Commit.scala]:
```suggestion
  }
```

**Review by conradoverta — COMMENTED** [review]
Minor changes

File: client/scala/src/main/scala/ai/verta/repository/Commit.scala (hunk 1 above)
Comment (nhatsmrt) [reply client/scala/src/main/scala/ai/verta/repository/Commit.scala]: Fixed

File: client/scala/src/main/scala/ai/verta/repository/Commit.scala (hunk 2 above)
Comment (nhatsmrt) [reply client/scala/src/main/scala/ai/verta/repository/Commit.scala]: Fixed

**Review by conradoverta — APPROVED** [review]

**Pull request merged**